### PR TITLE
Run2-alca110 Modify two plugins to conform standard

### PR DESCRIPTION
**Pull request by bsunanda**
IsolatedParticlesGeneratedJets and IsolatedTracksCone are modified to conform current standards

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 1]
```diff
@@ -1,48 +1,323 @@
...
+
+  double minTrackP_, maxTrackEta_, maxNearTrackP_;
+  
+  int    debugEcalSimInfo_;
+
+  bool   applyEcalIsolation_;
+
+  // track associator to detector parameters 
+  TrackAssociatorParameters parameters_;
+  mutable TrackDetectorAssociator* trackAssociator_;
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: why is this required to be `mutable`?

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 2]
```diff
@@ -1,48 +1,323 @@
...
+};
+
 IsolatedTracksCone::IsolatedTracksCone(const edm::ParameterSet& iConfig) :
    trackerHitAssociatorConfig_(consumesCollector()) {
 
   //now do what ever initialization is needed
-  doMC            = iConfig.getUntrackedParameter<bool>  ("DoMC", false); 
-  myverbose_      = iConfig.getUntrackedParameter<int>( "Verbosity", 5 );
+  doMC_            = iConfig.getUntrackedParameter<bool>  ("DoMC", false); 
+  myverbose_      = iConfig.getUntrackedParameter<int>( "Verbosity", 5);
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: please use lower case for the first letter of python configuration parameter names

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 3]
```diff
@@ -73,26 +348,31 @@ IsolatedTracksCone::IsolatedTracksCone(const edm::ParameterSet& iConfig) :
 
   edm::ParameterSet parameters = iConfig.getParameter<edm::ParameterSet>("TrackAssociatorParameters");
   edm::ConsumesCollector iC = consumesCollector();
-  parameters_.loadParameters( parameters, iC );
+  parameters_.loadParameters( parameters, iC);
   trackAssociator_ =  new TrackDetectorAssociator();
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: not part of this PR, but you could use a unique_ptr for the `TrackDetectorAssociator`

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 4]
```diff
@@ -73,26 +348,31 @@ IsolatedTracksCone::IsolatedTracksCone(const edm::ParameterSet& iConfig) :
...
 IsolatedTracksCone::~IsolatedTracksCone() {
   delete  trackAssociator_;
 }
 
+void IsolatedTracksCone::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
+}
+
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: what about a non-default parameter set description?

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 5]
```diff
@@ -166,14 +446,14 @@ void IsolatedTracksCone::analyze(const edm::Event& iEvent,
...
-  const CaloSubdetectorGeometry* gHE = 
-    geo->getSubdetectorGeometry(DetId::Hcal,HcalEndcap);
+  CaloSubdetectorGeometry* gEB = (CaloSubdetectorGeometry*)
+    (geo->getSubdetectorGeometry(DetId::Ecal,EcalBarrel));
+  CaloSubdetectorGeometry* gEE = (CaloSubdetectorGeometry*)
+    (geo->getSubdetectorGeometry(DetId::Ecal,EcalEndcap));
+  CaloSubdetectorGeometry* gHB = (CaloSubdetectorGeometry*)
+    (geo->getSubdetectorGeometry(DetId::Hcal,HcalBarrel));
+  CaloSubdetectorGeometry* gHE = (CaloSubdetectorGeometry*)
+    (geo->getSubdetectorGeometry(DetId::Hcal,HcalEndcap));
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: Why are casts required here?
Why removing `const`?
Why using C-style casts?

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 6]
```diff
@@ -334,12 +614,12 @@ void IsolatedTracksCone::analyze(const edm::Event& iEvent,
       nFailHighPurityQaul++;
     }
     
-    hRawPt ->Fill(pt1 );
-    hRawP  ->Fill(p1  );
-    hRawEta->Fill(eta1);
-    hRawPhi->Fill(phi1);
+    h_RawPt ->Fill(pt1);
+    h_RawP  ->Fill(p1 );
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: minor comment, but since you are anyway making a cosmetic change here, you could also remove the remaining whitespace in `(p1 )`

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 7]
```diff
@@ -929,15 +1207,15 @@ void IsolatedTracksCone::beginJob() {
...
   
-  double tempgen_TH[22] = { 0.0,  1.0,  2.0,  3.0,  4.0,  
-			    5.0,  6.0,  7.0,  8.0,  9.0, 
-			    10.0, 12.0, 15.0, 20.0, 25.0, 
-			    30.0, 40.0, 50.0, 60.0, 70.0, 80.0, 100};
-  for(int i=0; i<22; i++)  genPartPBins[i]  = tempgen_TH[i];
+  double tempgen_TH[nPBins_+1] = { 0.0,  1.0,  2.0,  3.0,  4.0,  
+				   5.0,  6.0,  7.0,  8.0,  9.0, 
+				   10.0, 12.0, 15.0, 20.0, 25.0, 
+				   30.0, 40.0, 50.0, 60.0, 70.0, 80.0, 100};
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: I think here and also above you could replace this with `std::array` and just use the nice interface for iterating trough it

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 8]
```diff
@@ -1080,7 +1358,7 @@ void IsolatedTracksCone::beginJob() {
   t_ievt = new std::vector<unsigned int>();
   t_ilum = new std::vector<unsigned int>();
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: not part of this PR, but these vectors are never deleted

**Review by ghellwig — CHANGES_REQUESTED** [review]

**Comment by davidlange6** [comment]
please test

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc (hunk 5 above)
Comment (bsunanda) [reply Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: This is corrected in a separate PR

Comment (bsunanda) [reply Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: I do not want to make this change to have conflict in PR's

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc (hunk 8 above)
Comment (bsunanda) [reply Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: These are part of a branch of a Tree. I am not sure if these to be deleted.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
The error is due to DAS not finding a data set. not related to this PR

**Comment by bsunanda** [comment]
@cmsbuild please test

**Comment by bsunanda** [comment]
@ghellwig Can you please approve this PR?

**Comment by bsunanda** [comment]
@lpernie Can you approve this PR?

File: Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc [hunk 9]
```diff
@@ -466,6 +478,10 @@ void RecAnalyzerMinbias::analyzeHcal(const HBHERecHitCollection & HithbheMB,
 	if (itr1 != histHC_.end()) itr1->second->Fill(energyhit);
       }
       h_[hid.subdet()-1]->Fill(energyhit);
+      if(energyhit >2) {
```
[the remark this answers is not in the document]
Comment (ghellwig) [reply Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc]: @bsunanda could you please comment on this?

[the remark this answers is not in the document]
Comment (ghellwig) [reply Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc]: @bsunanda nevermind, I looked at an outdated version and found that you answered in the code

File: Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc [hunk 10]
```diff
@@ -107,6 +127,9 @@ RecAnalyzerMinbias::RecAnalyzerMinbias(const edm::ParameterSet& iConfig) :
   eHighHE_              = iConfig.getParameter<double>("EHighHE");
   eLowHF_               = iConfig.getParameter<double>("ELowHF");
   eHighHF_              = iConfig.getParameter<double>("EHighHF");
+  eMin_                 = iConfig.getUntrackedParameter<double>("EMin",2.0);
+  runMin_               = iConfig.getUntrackedParameter<double>("RunMin",303441.5);
+  runMax_               = iConfig.getUntrackedParameter<double>("RunMax",304825.5);
```
Comment (ghellwig) [inline Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc]: Where do these default values come from?
Does it make sense to have doubles instead of integers as run number?

File: Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc [hunk 11]
```diff
@@ -173,73 +196,104 @@ RecAnalyzerMinbias::RecAnalyzerMinbias(const edm::ParameterSet& iConfig) :
...
+  desc.add<bool>("Noise",     false);
+  desc.add<double>("ELowHB",  4);
+  desc.add<double>("EHighHB", 100);
+  desc.add<double>("ELowHE",  4);
+  desc.add<double>("EHighHE", 150);
+  desc.add<double>("ELowHF",  10);
+  desc.add<double>("EHighHF", 150);
+  desc.addUntracked<double>("EMin",2.0);
+  desc.addUntracked<double>("RunMin",303441.5);
+  desc.addUntracked<double>("RunMax",304825.5);
```
Comment (ghellwig) [inline Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc]: as above, a short comment on how these numbers were obtained would be good for future users of the code

File: Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc [hunk 12]
```diff
@@ -173,73 +196,104 @@ RecAnalyzerMinbias::RecAnalyzerMinbias(const edm::ParameterSet& iConfig) :
...
-	mom0_MB  = info.theMB0;
+ void RecAnalyzerMinbias::beginJob() {
+
+   std::string hc[5] = {"Empty", "HB", "HE", "HO", "HF"};
+   char        name[700], title[700];
+   hbhe_ = fs_->make<TH2D>("hbhe","Noise in HB/HE",61,-30.5,30.5,72,0.5,72.5);
+   hb_   = fs_->make<TH2D>("hb",  "Noise in HB",61,-16.5,16.5,72,0.5,72.5);
+   he_   = fs_->make<TH2D>("he",  "Noise in HE",61,-30.5,30.5,72,0.5,72.5);
+   hf_   = fs_->make<TH2D>("hf",  "Noise in HF",82,-41.5,41.5,72,0.5,72.5);
+   int nbin = (int)(runMax_-runMin_+0.2);
```
Comment (ghellwig) [inline Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc]: No C-style cast, please.
This also raises again the question of the usefulness of doubles as run numbers.

File: Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc [hunk 13]
```diff
@@ -173,73 +196,104 @@ RecAnalyzerMinbias::RecAnalyzerMinbias(const edm::ParameterSet& iConfig) :
...
+   herun_   = fs_->make<TProfile>("herun",title,nbin,runMin_,runMax_,0.0,1.0);
+   sprintf (title, "Fraction of channels in HF with E > %4.1f GeV vs Run number", eMin_);
+   hfrun_   = fs_->make<TProfile>("hfrun",title,nbin,runMin_,runMax_,0.0,1.0);
+   for(int idet=1; idet<=4; idet++){
+     sprintf(name, "%s", hc[idet].c_str());
+     sprintf (title, "Noise distribution for %s", hc[idet].c_str());
+     h_[idet-1] = fs_->make<TH1D>(name,title,48,-6., 6.);
+   }
+
+   for (unsigned int i=0; i<hcalID_.size(); i++) {
```
Comment (ghellwig) [inline Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc]: If I am not mistaken you use `i` only to get the items in `hcalID_`, i.e. why not replacing this with a range-based loop:
```
for (const auto& hcalID: hcalID_) {
...
```

File: Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc [hunk 14]
```diff
@@ -173,73 +196,104 @@ RecAnalyzerMinbias::RecAnalyzerMinbias(const edm::ParameterSet& iConfig) :
...
+   myMap_.clear();
+ }
+
+ //  EndJob
+ //
+ void RecAnalyzerMinbias::endJob() {
+
+   if (!fillHist_) {
+     cells = 0;
+     for (std::map<std::pair<int,HcalDetId>,myInfo>::const_iterator itr=myMap_.begin(); itr != myMap_.end(); ++itr) {
```
Comment (ghellwig) [inline Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc]: a range-based loop would be better here, too

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 15]
```diff
@@ -1,96 +1,594 @@
...
+  delete t_v_RH_h5x5_ene ;
+  delete t_v_RH_r26_ieta ;
+  delete t_v_RH_r26_iphi ;
+  delete t_v_RH_r26_ene  ;
+  delete t_v_RH_r44_ieta ;
+  delete t_v_RH_r44_iphi ;
+  delete t_v_RH_r44_ene  ;
+  delete t_irun;
+  delete t_ievt;
+  delete t_ilum;
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: If I am not mistaken, these could all become `unique_ptr`s (with the appropriate changes everywhere)

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc [hunk 16]
```diff
@@ -1,96 +1,594 @@
...
+  desc_TrackAssoc.add<edm::InputTag>("EBRecHitCollectionLabel",edm::InputTag("ecalRecHit","EcalRecHitsEB"));
+  desc_TrackAssoc.add<double>("dRMuonPreselection",0.2);
+  desc_TrackAssoc.add<bool>("truthMatch",false);
+  desc_TrackAssoc.add<edm::InputTag>("HBHERecHitCollectionLabel",edm::InputTag("hbhereco"));
+  desc_TrackAssoc.add<bool>("useHcal",true);
+  desc_TrackAssoc.add<bool>("usePreshower",false);
+  desc_TrackAssoc.add<double>("dRPreshowerPreselection",0.2);
+  desc_TrackAssoc.add<double>("trajectoryUncertaintyTolerance",1.0);
+  desc.add<edm::ParameterSetDescription>("TrackAssociatorParameters",desc_TrackAssoc);
+  descriptions.add("isolatedTracksCone",desc);
```
Comment (ghellwig) [inline Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: please comment on the default parameters

Comment (bsunanda) [reply Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: Cannot provide justification - they were used as default while evaluating the use of IsoTracks in calibration

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc (hunk 16 above)
Comment (ghellwig) [reply Calibration/IsolatedParticles/plugins/IsolatedTracksCone.cc]: Ok, I think that's fine then

**Comment by ghellwig** [comment]
+1

**Comment by fabiocos** [comment]
+1
